app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeViewModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.wire.android.ui.authentication.welcome
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
+import com.wire.android.navigation.NavigationManager
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class WelcomeViewModel @Inject constructor(
+    private val navigationManager: NavigationManager,
+) : ViewModel() {
+
+    fun navigateBack() {
+        viewModelScope.launch {
+            navigationManager.navigateBack()
+        }
+    }
+
+    fun onLoginPressed() {
+        navigate(NavigationCommand(NavigationItem.Login.getRouteWithArgs()))
+    }
+
+    fun onCreateEnterpriseAccountPressed() {
+        navigate(NavigationCommand(NavigationItem.CreateEnterpriseAccount.getRouteWithArgs()))
+    }
+
+    fun onCreatePrivateAccountPressed() {
+        navigate(NavigationCommand(NavigationItem.CreatePrivateAccount.getRouteWithArgs()))
+    }
```

Comment:
in my opinion the names from the viewmodel should not indicate the "view" action but just a action like "navigateToLogin", "goToEnterPriseAccount"